exec/plugin-commands-script-runners/src/dlx.ts

```diff
@@ -91,7 +91,17 @@ export async function handler (
       saveOptional: false,
       savePeer: false,
     }, pkgs)
-    await symlinkDir(cachedDir, cacheLink, { overwrite: true })
+    try {
+      await symlinkDir(cachedDir, cacheLink, { overwrite: true })
+    } catch (error) {
+      // EBUSY means that there is another dlx process running in parallel that has acquired the cache link first.
+      // The current process should yield.
+      if (util.types.isNativeError(error) && 'code' in error && error.code === 'EBUSY') {
+        await new Promise(resolve => setTimeout(resolve, 0))
+      } else {
+        throw error
+      }
+    }
```

Comment:
it should retry, no? Also, this change probably should be done in the `symlink-dir` library.

Reply:
> it should retry, no?

The first process should already create the cache link, which is just as up-to-date as the link this process was trying to create. I don't think we need to create it again.

> Also, this change probably should be done in the `symlink-dir` library.

`retries` would make sense in `symlink-dir`. But what about skipping on `EBUSY`? It's very situational.

Reply:
ok, just add more details to the comment explaining why the symlink created by the other process is fine.

Reply:
https://github.com/pnpm/pnpm/pull/8604/commits/e94a1a86c525ba2aded89d850f43e1a193d71224